### ISSUE TITLE
Wrong timezone used for DateTimeImmutable

### ISSUE THREAD

**Issue by checkfrontChase** [slope-it/clock-mock#34]
When ClockMock is active, DateTimeImmutable will not use the desired type 3 timezone. The timezone seems to always change to a type 2 timezone. This does not affect DateTime.

Example:
```
$date = new DateTimeImmutable('2023-03-01');
echo $date->getTimezone()->getName();      // 'Canada/Central'

ClockMock::freeze($date);

$date = new DateTimeImmutable('2023-03-01');
echo $date->getTimezone()->getName();      // 'CDT'

$date = new DateTime('2023-03-01');
echo $date->getTimezone()->getName();      // 'Canada/Central'
```

**Comment by checkfrontChase** [comment]
Changing the format used in `DateTimeImmutableMock` seems to resolve this. Instead of using `Y-m-d\TH:i:s.uT`, it could use `Y-m-d\TH:i:s.ue`.

```
class DateTimeImmutableMock extends \DateTimeImmutable
{
    public function __construct(?string $datetime = 'now', ?DateTimeZone $timezone = null)
    {
        // Create an immutable instance starting from the mutable mock, so we don't have to replicate mocking logic.
        $mutableDateTime = new DateTimeMock($datetime, $timezone);

        parent::__construct($mutableDateTime->format('Y-m-d\TH:i:s.ue'), $timezone);
    }
}
```
`e` will give the "timezone identifier", whereas `T` always gives the abbreviation or UTC offset.

**Comment by asprega** [comment]
Closing as fixed via PR #35

**Comment by asprega** [comment]
Released in 0.3.6

Thanks @checkfrontChase for reporting it and @holmberd for fixing it!
